src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -197,39 +248,42 @@ internal sealed override ImmutableArray<string> NotNullWhenFalseMembers
             }
         }
 
+#nullable enable
         private SourcePropertyAccessorSymbol(
             NamedTypeSymbol containingType,
             string name,
-            SourcePropertySymbol property,
+            SourcePropertySymbolBase property,
             DeclarationModifiers propertyModifiers,
             ImmutableArray<MethodSymbol> explicitInterfaceImplementations,
             Location location,
-            AccessorDeclarationSyntax syntax,
+            CSharpSyntaxNode syntax,
+            bool hasBody,
+            bool hasExpressionBody,
+            SyntaxTokenList modifiers,
             MethodKind methodKind,
+            bool usesInit,
             bool isAutoPropertyAccessor,
             bool isExplicitInterfaceImplementation,
             DiagnosticBag diagnostics)
             : base(containingType,
                    syntax.GetReference(),
                    location,
-                   isIterator: SyntaxFacts.HasYieldOperations(syntax.Body))
+                   isIterator: SyntaxFacts.HasYieldOperations(syntax))
```

Comment:
>SyntaxFacts.HasYieldOperations(syntax) [](start = 31, length = 38)

This looks fragile, we don't know what syntax node we are given and whether we can treat it as a body. Let's pass isIterator as a parameter.